applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+            
+        
+        // TODO: Probably could use better dependecies, but if it works it works
+        const getSelectedClasses = useCallback(() => {
+            const selectedClasses = Object.entries(connectionsMap).map(([k, v]) => {                
+                if(v) {
+                    return k;
+                }
+            });
+            if(selectedClasses === undefined) {
+                return [];
+            }
+            else {
+                return selectedClasses.filter(e => e !== undefined);
+            }
+        }, [connectionsMap]);
+
+        const onAcceptCallback = useCallback(() => {
+            if(onSelectConfirmCallback === undefined) {                
+                close();                
+            }
+            else {
+                onSelectConfirmCallback(getSelectedClasses());
+                close();
+            }            
+        }, [getSelectedClasses]);
+           
+
+        // TODO: Probably will have to show profiles later differently (like in catalog - using TreeLikeOffset)
+        return (
+            <BaseDialog heading={t("select-classes-dialog.select-classes")}>
+                <div>
+                    {/* We could also modify and then reuse <EntityCatalog />, it would be kind of nice, but it would take a lot of time */}
+                    <ul>
+                        {[...models.entries()].map(([modelId, model]) => 
+                            <div key={`select-classes-dialog-div-${model.getId()}`}>
+                                <h4 key={`select-classes-dialog-h4-${model.getId()}`}>Ⓜ {model.getAlias() ?? ""}</h4>
+                                {Object.values(model.getEntities()).map(e => {
+                                    if(isSemanticModelClass(e) || isSemanticModelClassUsage(e)) {
+                                        return <div key={`select-classes-dialog-div-${e.id}`} 
+                                                    style={{ backgroundColor: tailwindColorToHex(activeVisualModel?.getColor(modelId) ?? getDefaultColor()) }}>
+                                                    <li key={`select-classes-dialog-li-${e.id}`}><input type="checkbox" 
+                                                        onChange={(event) => setConnectionsMap(createChangedRecord(connectionsMap, e.id, event.target.checked))}
+                                                        // Note that if it is undefined we also set the value to false.
+                                                        // We do this because otherwise the order of the created associations is same as the chosen order and not the model order
+                                                        checked={connectionsMap[e.id] === undefined ? (connectionsMap[e.id] = false) : connectionsMap[e.id]}></input> {getPreferredName(e)}</li>
+                                            </div>;
+                                    }
+                                })
```

Comment:
Also you render only when the condition pass otherwise you have no return. 
It would be better to filter the values in the first place. 
You can store the filtered values in a variable: options, availableClassesOrClassUsages .. etc ..